src/features/ci-cd/hooks/device-update-trigger.ts

```diff
@@ -25,3 +17,62 @@ hooks.addPureHook('PATCH', 'resin', 'device', {
 		}
 	},
 });
+
+hooks.addPureHook('POST', 'resin', 'device_application', {
+	/**
+	 * Notify device to update if they've had an app added
+	 * TODO-MULTI-APP: Deleting apps is also covered by this in single-multi-app world but needs to be handled in true multi-app
+	 */
+	POSTRUN: async ({ request, result }) => {
+		if (result == null) {
+			return;
+		}
+
+		await postDevices({
+			url: '/v1/update',
+			req: permissions.root,
+			filter: { id: request.values.device },
+			// Don't wait for the post to complete, as it may take a long time
+			wait: false,
+		});
+	},
+});
+
+hooks.addPureHook('PATCH', 'resin', 'device_application', {
+	POSTRUN: async (args) => {
+		const affectedIds = args.request.affectedIds!;
+		if (
+			affectedIds.length === 0 ||
+			args.request.values.should_be_running__release === undefined
```

Comment:
Do we also need to handle `device_application` PATCHes of the app property?

Reply:
Yes and no, they aren't mean to be allowed to be updated so doing so is a mistake - technically open-balena-api grants much more access so it is possible but doing so is not intended and would ideally be blocked